lib/pages/chat_list/chat_list_view_builder.dart

```diff
@@ -21,49 +25,112 @@ class ChatListViewBuilder extends StatelessWidget {
       physics: const NeverScrollableScrollPhysics(),
       itemCount: rooms.length,
       itemBuilder: (BuildContext context, int index) {
-        return ValueListenableBuilder(
+        return ValueListenableBuilder<SelectMode>(
           valueListenable: controller.selectModeNotifier,
-          builder: (context, _, __) {
-            return ValueListenableBuilder(
-              valueListenable: controller.widget.activeRoomIdNotifier,
-              builder: (context, activeRoomId, child) {
-                return ChatListItem(
-                  rooms[index],
-                  key: Key('chat_list_item_${rooms[index].id}'),
-                  isEnableSelectMode: controller.isSelectMode,
-                  onTap: controller.isSelectMode
-                      ? () => controller.toggleSelection(rooms[index].id)
-                      : null,
-                  onSecondaryTap: () => controller.handleContextMenuAction(
-                    context,
-                    rooms[index],
-                  ),
-                  onLongPress: () => controller.onLongPressChatListItem(
-                    rooms[index],
-                  ),
-                  checkBoxWidget: ValueListenableBuilder(
-                    valueListenable: controller.conversationSelectionNotifier,
-                    builder: (context, conversationSelection, __) {
-                      final conversation =
-                          conversationSelection.firstWhereOrNull(
-                        (conversation) =>
-                            conversation.roomId.contains(rooms[index].id),
-                      );
-                      return Checkbox(
-                        value: conversation?.isSelected == true,
-                        onChanged: (_) {
-                          controller.toggleSelection(rooms[index].id);
-                        },
-                      );
-                    },
-                  ),
-                  activeChat: activeRoomId == rooms[index].id,
-                );
-              },
-            );
+          builder: (context, selectMode, __) {
+            return ChatListViewStyle.responsiveUtils
+                        .isMobileOrTablet(context) &&
+                    selectMode == SelectMode.normal
+                ? _SlidableChatListItem(
+                    controller: controller,
+                    room: rooms[index],
+                    chatListItem: _CommonChatListItem(
+                      controller: controller,
+                      room: rooms[index],
+                    ),
+                  )
+                : _CommonChatListItem(
+                    controller: controller,
+                    room: rooms[index],
+                  );
```

Comment:
```dart
ValueListenableBuilder(
          valueListenable: controller.selectModeNotifier,
          builder: (context, selectMode, __) {
            if (ChatListViewStyle.responsiveUtils
                .isMobileOrTablet(context) &&
                selectMode.isSelectMode) {
              _SlidableChatListItem(
                controller: controller,
                room: rooms[index],
                chatListItem: _CommonChatListItem(
                  controller: controller,
                  room: rooms[index],
                ),
              )
            }
          },
          child: _CommonChatListItem(
            controller: controller,
            room: rooms[index],
          ),
        );
```

in `lib/presentation/enum/chat_list/chat_list_enum.dart`

```dart
enum SelectMode {
  normal,
  select;

  bool get isSelectMode => this == SelectMode.select;
}
```

Reply:
I like the isSelectMode but the child part adds nothing but more complexity

Reply:
```dart
       ValueListenableBuilder(
          valueListenable: controller.selectModeNotifier,
          builder: (context, selectMode, __) {
            if (ChatListViewStyle.responsiveUtils
                .isMobileOrTablet(context) &&
                selectMode.isSelectMode) {
              _SlidableChatListItem(
                controller: controller,
                room: rooms[index],
                chatListItem: _CommonChatListItem(
                  controller: controller,
                  room: rooms[index],
                ),
              )
            }
          return child!;
          },
          child: _CommonChatListItem(
            controller: controller,
            room: rooms[index],
          ),
        );
```

Reply:
I want to show _CommonChatListItem when `
ChatListViewStyle.responsiveUtils
                .isMobileOrTablet(context) &&
                selectMode.isSelectMode
` is false.
I don't see the value in your proposal, it adds nothing more to my current code

Reply:
I'd prefer to do that then
```dart
       ValueListenableBuilder(
          valueListenable: controller.selectModeNotifier,
          builder: (context, selectMode, __) {
            if (ChatListViewStyle.responsiveUtils
                .isMobileOrTablet(context) &&
                selectMode.isSelectMode) {
              _SlidableChatListItem(
                controller: controller,
                room: rooms[index],
                chatListItem: child!,
              )
            }
          return child!;
          },
          child: _CommonChatListItem(
            controller: controller,
            room: rooms[index],
          ),
        );
```